### ISSUE TITLE
False positive on CA1508

### ISSUE THREAD

**Issue by jmarolf** [dotnet/roslyn-analyzers#4982]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1508): `Avoid dead conditional code`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers/5.0.3)

**Version**: 5.0.3 (Latest)

### Describe the bug

False positive on dead code analysis

### Steps To Reproduce

```csharp
public interface IArgument
{
    /// <summary>
    /// The value of the argument sans quotes.
    /// </summary>
    string Value { get; }
}

public interface IArguments : IReadOnlyList<IArgument>, IArgument
{
  void Deconstruct(out IArgument first, out IArgument second, out IArgument third);
}

var (verb, mention, description) = messageContext.Arguments; // messageContext.Arguments implements IArguments

// Error CA1508 : 'mention.Value is "not"' is always 'false'. Remove or refactor the condition(s) to avoid dead code.
if (verb.Value is "is" && mention.Value is "not")
{
}
```

```xml
<TargetFramework>net5.0</TargetFramework>
<Nullable>enable</Nullable>
<LangVersion>9.0</LangVersion>

...
<PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="5.0.3">
  <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
  <PrivateAssets>all</PrivateAssets>
</PackageReference>
```

### Expected behavior

No warning

## Additional context

This pattern is not flagged by the analyzer

```csharp
if (verb is {Value: "is"} && mention is {Value: "not" })
{
}
```
